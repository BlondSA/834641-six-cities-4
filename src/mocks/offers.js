export const offers = [
  {
    features: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`,
    ],
    bedrooms: 2,
    descriptionOffer: [
      `A quiet cozy and picturesque that hides behind a a river by
      the unique lightness of Amsterdam. The building is green and
      from 18th century.`,
      `An independent House, strategically located between Rembrand
      Square and National Opera, but where the bustle of the city
      comes to rest in this alley flowery and colorful.`,
    ],
    hostName: `George`,
    isHostPro: false,
    srcHostAvatar: `img/avatar-angelina.jpg`,
    id: 1,
    isInBookmark: false,
    isPremium: true,
    maxAdults: 3,
    price: 120,
    rating: 1,
    srcImageOffer: [
      `http://picsum.photos/260/200?r=4`,
      `http://picsum.photos/260/200?r=1`,
      `http://picsum.photos/260/200?r=2`,
      `http://picsum.photos/260/200?r=3`,
    ],
    srcPreviewImageOffer: `http://picsum.photos/260/200?r=1`,
    title: `Beautiful & luxurious apartment at great location`,
    type: `Apartment`,
    location: {
      latitude: 52.3909553943508,
      longitude: 4.85309666406198,
      zoom: 8,
    },
    city: {
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10,
      },
      name: `Amsterdam`,
    },
  },
  {
    features: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`,
    ],
    bedrooms: 1,
    descriptionOffer: [
      `A quiet cozy and picturesque that hides behind a a river by
      the unique lightness of Amsterdam. The building is green and
      from 18th century.`,
      `An independent House, strategically located between Rembrand
      Square and National Opera, but where the bustle of the city
      comes to rest in this alley flowery and colorful.`,
    ],
    hostName: `Angelina`,
    isHostPro: true,
    srcHostAvatar: `img/avatar-angelina.jpg`,
    id: 2,
    isInBookmark: true,
    isPremium: false,
    maxAdults: 3,
    price: 80,
    rating: 2,
    srcImageOffer: [
      `http://picsum.photos/260/200?r=1`,
      `http://picsum.photos/260/200?r=2`,
    ],
    title: `Wood and stone place`,
    type: `Private room`,
    location: {
      latitude: 52.369553943508,
      longitude: 4.85309666406198,
      zoom: 8,
    },
    city: {
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10,
      },
      name: `Amsterdam`,
    },
  },
  {
    features: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`,
    ],
    bedrooms: 2,
    descriptionOffer: [
      `A quiet cozy and picturesque that hides behind a a river by
      the unique lightness of Amsterdam. The building is green and
      from 18th century.`,
      `An independent House, strategically located between Rembrand
      Square and National Opera, but where the bustle of the city
      comes to rest in this alley flowery and colorful.`,
    ],
    hostName: `Ivan`,
    isHostPro: true,
    srcHostAvatar: `img/avatar-angelina.jpg`,
    id: 3,
    isInBookmark: false,
    isPremium: false,
    maxAdults: 3,
    price: 132,
    rating: 3,
    srcImageOffer: [
      `http://picsum.photos/260/200?r=2`,
    ],
    title: `Canal View Prinsengracht`,
    type: `Apartment`,
    location: {
      latitude: 52.3909553943508,
      longitude: 4.929309666406198,
      zoom: 8,
    },
    city: {
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10,
      },
      name: `Amsterdam`,
    },
  },
  {
    features: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
    ],
    bedrooms: 4,
    descriptionOffer: [
      `A quiet cozy and picturesque that hides behind a a river by
      the unique lightness of Amsterdam. The building is green and
      from 18th century.`,
      `An independent House, strategically located between Rembrand
      Square and National Opera, but where the bustle of the city
      comes to rest in this alley flowery and colorful.`,
    ],
    hostName: `Alexsandr`,
    isHostPro: true,
    srcHostAvatar: `img/avatar-angelina.jpg`,
    id: 4,
    isInBookmark: false,
    isPremium: true,
    maxAdults: 5,
    price: 180,
    rating: 90,
    srcImageOffer: [
      `http://picsum.photos/260/200?r=3`,
    ],
    title: `Nice, cozy, warm big bed apartment`,
    type: `Apartment`,
    location: {
      latitude: 52.3809553943508,
      longitude: 4.939309666406198,
      zoom: 8,
    },
    city: {
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10,
      },
      name: `Amsterdam`,
    },
  },
];
