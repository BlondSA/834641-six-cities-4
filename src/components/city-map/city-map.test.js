import React from 'react';
import renderer from 'react-test-renderer';
import CityMap from './city-map.jsx';
import {offers} from '../../mocks/offers.js';

describe(`Testing <CityMap/>`, () => {
  it(`<CityMap/> have rendered correctly show data`, () => {
    const tree = renderer
      .create(<CityMap offers={offers} city={`Amsterdam`} />, {
        createNodeMock: () => {
          return document.createElement(`div`);
        },
      })
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`without_data`, () => {
    const tree = renderer
      .create(<CityMap offers={[]} city={`Amsterdam`} />, {
        createNodeMock: () => {
          return document.createElement(`div`);
        },
      })
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
