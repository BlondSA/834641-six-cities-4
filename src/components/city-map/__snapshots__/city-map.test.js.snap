// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <CityMap/> <CityMap/> have rendered correctly show data 1`] = `
<section
  className="cities__map map"
/>
`;
